src/org/opensolaris/opengrok/analysis/AnalyzerGuru.java

```diff
@@ -465,13 +466,24 @@ public void populateDocument(Document doc, File file, String path,
             doc.add(npstring);
         }
 
+        StreamSource src = StreamSource.fromFile(file);
+        byte[] digest;
+        try (DigestedInputStream digestStream = src.getSHA256stream()) {
+            digest = digestStream.digestAll();
+        } catch (IOException ex) {
+            throw ex;
+        } catch (Exception ex) {
+            throw new IOException("DigestedInputStream close()");
+        }
+        doc.add(new StoredField(QueryBuilder.SHA256, digest));
```

Comment:
`StreamSource.SHA256` or better `StreamSource.getDigestName()` ?

Reply:
This is a key name into the Lucene document, which I believe could be argued is the purview of `QueryBuilder` (for historical reasons). Extracting an entirely new static class from `QueryBuilder` is one future idea, but it seems questionable whether it would be a good idea to disperse key names around the source code as with e.g. `StreamSource.SHA256` or `StreamSource.getDigestName()`.

Reply:
Can the key name be named in generic way then ? (`digest` ?)